app/navigation/BusinessSettlorNavigator.scala

```diff
@@ -36,46 +37,70 @@ class BusinessSettlorNavigator @Inject()() extends Navigator {
+    case CountryOfResidencePage => ua =>
+      navigateAwayFromResidencePages(mode, trustType, ua)
+    case AddressYesNoPage => ua =>
+      yesNoNav(ua, AddressYesNoPage, rts.LiveInTheUkYesNoController.onPageLoad(mode), navigateToEndPages(mode, trustType, ua))
     case LiveInTheUkYesNoPage => ua =>
       yesNoNav(ua, LiveInTheUkYesNoPage, rts.UkAddressController.onPageLoad(mode), rts.NonUkAddressController.onPageLoad(mode))
+    case UkAddressPage | NonUkAddressPage => ua =>
+      navigateToEndPages(mode, trustType, ua)
+    case CompanyTimePage => ua =>
+      if (mode == NormalMode) {
+        rts.StartDateController.onPageLoad()
+      } else {
+        checkDetailsRoute(ua)
+      }
   }
 
-  private def navigationWithCheckAndTrustType(mode: Mode, trustType: TypeOfTrust): PartialFunction[Page, UserAnswers => Call] = {
-    mode match {
-      case NormalMode => {
-        case UtrPage | UkAddressPage | NonUkAddressPage => _ =>
-          trustType match {
-            case EmployeeRelated =>
-              rts.CompanyTypeController.onPageLoad(mode)
-            case _ =>
-              rts.StartDateController.onPageLoad()
-          }
-        case CompanyTimePage => _ =>
-          rts.StartDateController.onPageLoad()
-        case AddressYesNoPage => ua =>
-          yesNoNav(ua, AddressYesNoPage, rts.LiveInTheUkYesNoController.onPageLoad(mode), rts.StartDateController.onPageLoad())
-      }
-      case CheckMode => {
-        case UtrPage | UkAddressPage | NonUkAddressPage => ua =>
-          trustType match {
-            case EmployeeRelated =>
-              rts.CompanyTypeController.onPageLoad(mode)
-            case _ =>
-              checkDetailsRoute(ua)
-          }
-        case CompanyTimePage => ua =>
-          checkDetailsRoute(ua)
-        case AddressYesNoPage => ua =>
-          yesNoNav(ua, AddressYesNoPage, rts.LiveInTheUkYesNoController.onPageLoad(mode), checkDetailsRoute(ua))
-      }
+
+  private def navigateAwayFromNamePage(mode: Mode, answers: UserAnswers): Call = {
+    if (answers.is5mldEnabled && !answers.isTaxable) {
+      rts.CountryOfResidenceYesNoController.onPageLoad(mode)
+    } else {
+      rts.UtrYesNoController.onPageLoad(mode)
+    }
+  }
+
+  private def navigateAwayFromUtrPages(mode: Mode, trustType: TypeOfTrust, answers: UserAnswers): Call = {
+    (answers.is5mldEnabled, answers.get(UtrYesNoPage)) match {
+      case (true, _) => rts.CountryOfResidenceYesNoController.onPageLoad(mode)
+      case (false, Some(true)) => navigateToEndPages(mode, trustType, answers)
+      case (false, _) => rts.AddressYesNoController.onPageLoad(mode)
+    }
+  }
+
+  private def navigateAwayFromResidencePages(mode: Mode, trustType: TypeOfTrust, answers: UserAnswers): Call = {
+    val hasUtr = answers.get(UtrYesNoPage).getOrElse(false)
+    val isNonTaxable5mld = answers.is5mldEnabled && !answers.isTaxable
```

Comment:
It is likely this logic will be used by the individual navigator as well. Consider making this a def in the Navigator trait, which line 73 can also use

Reply:
I'd rather not do this. I have no idea of what the next bit of logic will need. It's up to the next person to do the refactor.
i.e.
navigateAwayFromResidencePages calls one of about 4 different routes